cmd/wallet-web/src/mixins/gnap/store.js

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+const dbName = 'gnap-store';
+const dbObjectStore = 'gnap-keypair-store';
+
+export async function getGnapKeyPair() {
+  const dbExists = (await window.indexedDB.databases()).map((db) => db.name).includes(dbName);
+
+  if (!dbExists) {
+    return createAndStoreGnapKeyPair();
+  }
+
+  const gnapKeyPair = await fetchStoredGnapKeyPair();
+  return gnapKeyPair;
+}
+
+async function createAndStoreGnapKeyPair() {
+  const gnapKeyPair = await createKeyPair();
+  callOnStore(function (store) {
+    store.add(gnapKeyPair);
+  });
+  return gnapKeyPair;
+}
+
+function fetchStoredGnapKeyPair() {
+  return new Promise(function (resolve) {
+    callOnStore(function (store) {
+      store.get(1).onsuccess = function (event) {
+        return resolve(event.target.result);
+      };
+    });
+  });
+}
+
+export function clearGnapStoreData() {
+  const req = indexedDB.deleteDatabase(dbName);
+  req.onsuccess = function () {
+    console.log('Deleted database successfully', dbName);
+  };
+  req.onerror = function () {
+    console.log("Couldn't delete database");
+  };
+}
+
+function callOnStore(fn_) {
+  // This works on all devices/browsers, and uses IndexedDBShim as a final fallback
+  const indexedDB =
+    window.indexedDB ||
+    window.mozIndexedDB ||
+    window.webkitIndexedDB ||
+    window.msIndexedDB ||
+    window.shimIndexedDB;
+
+  const open = indexedDB.open(dbName, 3);
+  console.log('coming in call on store', open);
+  // Create the schema
+  open.onupgradeneeded = function () {
+    const db = open.result;
+    const store = db.createObjectStore(dbObjectStore, { autoIncrement: true });
+  };
+
+  open.onsuccess = function () {
+    // Start a new transaction
+    const db = open.result;
+    const tx = db.transaction(dbObjectStore, 'readwrite');
+    const store = tx.objectStore(dbObjectStore);
+    fn_(store);
+    // Close the db when the transaction is done
+    tx.oncomplete = function () {
+      db.close();
+    };
+  };
+}
+
+// TODO Issue-364 Add gnap create key pair logic to agent sdk library
+function createKeyPair() {
```

Comment:
follow up: do not hardcode keytype and other parameters.

Reply:
Part of #364 Agent-sdk